lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -523,6 +548,59 @@ class ConvertAtenMaxPool2dWithIndicesBackwardOp
 };
 } // namespace
 
+namespace {
+class ConvertAtenCumsumOp : public OpConversionPattern<AtenCumsumOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenCumsumOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Value input = adaptor.self();
+    auto resultType = input.getType().cast<RankedTensorType>();
+    Type elementType = resultType.getElementType();
+    int64_t inputRank = resultType.getRank();
+    Location loc = op->getLoc();
+    Value dtype = op.dtype();
+    if (!dtype.getType().isa<Torch::NoneType>())
+      return rewriter.notifyMatchFailure(
+          op, "unsupported: dtype argument not supported");
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only constant dim value is supported");
+    if (!isValidDim(dim, inputRank))
+      dim = toPositiveDim(dim, inputRank);
```

Comment:
This should be done the other way. First do `toPositiveDim`, then check if it is valid. If it is not valid, a `notifyMatchFailure` should be returned with an error message. See for example:

https://github.com/llvm/torch-mlir/blob/99093d0623c3d3e6fb961ad0c8fda5ee9fc264fe/lib/Conversion/TorchToLinalg/DataMovement.cpp#L80-L82